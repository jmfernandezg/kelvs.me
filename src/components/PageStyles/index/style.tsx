import styled from "styled-components";

export const HeroWrapper = styled.div`
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  min-height: calc(100vh - 7rem);
  margin-top: 3em;
  ${({ theme }) => theme.desktop`
    justify-content: center;
    margin-bottom: 0;
    margin-top: 0;
    min-height: calc(65vh - 7rem);
    `}
`;

// TODO: put a div in the hero that can be tilted like in react spring
export const HeroOne = styled.div`
  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: flex-start;
  font-family: "Caveat", cursive;
  h1 {
    display: inline-block;
    overflow: visible;
    width: 100%;
    flex: 0 auto;
    font-size: 4em;
    font-weight: 400;
    text-align: center;
    ${({ theme }) => theme.thone`font-size: 2em;`}
  }
  span {
    font-weight: 700;
  }
`;

export const HeroTwo = styled.div`
  h1 {
    position: relative;
    font-size: 3.5em;
    font-weight: 400;
    text-align: center;
    line-height: 58px;
    font-family: "Lato", sans-serif;
    ${({ theme }) => theme.thone`font-size: 2em;`}
    ${({ theme }) => theme.phablet`
      font-size: 1.5em;
      line-height: 35px;
    `}
  }
  span {
    padding-right: 5px;
    padding-left: 5px;
    background: linear-gradient(
      transparent 10%,
      #495867 10%,
      #495867 100%,
      transparent 90%
    );
    color: #fff;
    font-weight: 300;
    text-align: justify;
  }
`;

export const CallToAction = styled.div`
  margin: 4em 0;
  font-size: 15px;
  ${({ theme }) => theme.tablet`margin: 2em 0 0 0;`}
  a {
    padding: 0 20px;
    height: 25px;
  }
  a::before,
  a::after {
    position: absolute;
    width: 25px;
    height: 1px;
    background: ${props => props.theme.text.dark};
    content: "";
    opacity: 0.2;
    transition: all 0.3s;
    pointer-events: none;
  }
  .dark-mode & a::before,
  .dark-mode & a::after {
    background: ${props => props.theme.text.light};
  }
  a:hover {
    opacity: 0.9;
  }
  a::before {
    top: 0;
    left: 0;
    transform: rotate(90deg);
    transform-origin: 0 0;
  }
  a::after {
    right: 0;
    bottom: 0;
    transform: rotate(90deg);
    transform-origin: 100% 0;
  }
  a:hover::before,
  a:hover::after,
  a:focus::before,
  a:focus::after {
    opacity: 1;
  }
  a:hover::before,
  a:focus::before {
    left: 50%;
    transform: rotate(0deg) translateX(-50%);
  }

  a:hover::after,
  a:focus::after {
    right: 50%;
    transform: rotate(0deg) translateX(50%);
  }
`;

export const Contact = styled.div`
  width: 20%;
  display: flex;
  justify-content: space-around;
  svg {
    height: 1.5em;
    width: 100%;
    fill: ${props => props.theme.text.dark};
  }
  .dark-mode & svg {
    fill: ${props => props.theme.text.light};
  }
  ${({ theme }) => theme.bigTablet`
    margin: 2em 0;
    svg {
      height: 1.1em;
    }
  `}
  ${({ theme }) => theme.thone`
    width: 40%;
  `}
`;

export const FeaturedProjects = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  align-items: center;
`;

export const AppBox = styled.div`
  background: #fff 0 0 no-repeat padding-box;
  box-shadow: 0 3px 6px rgba(0, 0, 0, 0.16);
  width: 32%;
  padding: 0.75em;
  margin-bottom: 1em;
  position: relative;
  transition: transform 0.2s;
  min-height: 290px;
  &:hover {
    transform: scale(1.025);
  }
  color: ${props => props.theme.text.dark};
  img {
    border: 1px solid #000;
    width: 100%;
    height: auto;
  }
  span {
    margin-right: 3px;
    background-color: #ff4e4e;
    color: #fff;
    padding: 0 3px 1px 3px;
    line-height: normal;
    font-size: 12px;
    font-weight: 300;
    letter-spacing: 1px;
    border-radius: 3px;
  }
  p {
    margin-bottom: 0;
  }
  ${({ theme }) => theme.tablet`
    width: 48%;
    p {
      font-size: 0.9rem;
    }
  `}
  ${({ theme }) => theme.thone`width: 100%;`}
`;

export const H1 = styled.h1`
  margin: 2.5em 0;
  ${({ theme }) => theme.tablet`
    margin: 2em 0;
    font-size: 1.125rem;
  `}
  font-weight: 700;
`;
