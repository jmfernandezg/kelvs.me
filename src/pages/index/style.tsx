import styled from "styled-components";

export const HeroWrapper = styled.div`
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  min-height: 40vh;
  ${({ theme }) => theme.tablet`justify-content: center;`}
`;

// TODO: put a div in the hero that can be tilted like in react spring
export const HeroOne = styled.div`
  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: flex-start;
  font-family: "Caveat", cursive;
  h1 {
    display: inline-block;
    overflow: visible;
    width: 100%;
    flex: 0 auto;
    font-size: 4em;
    font-weight: 400;
    text-align: center;
    ${({ theme }) => theme.thone`font-size: 2em;`}
  }
  span {
    font-weight: 700;
  }
`;

export const HeroTwo = styled.div`
  h1 {
    position: relative;
    font-size: 3.5em;
    font-weight: 400;
    text-align: center;
    line-height: 58px;
    font-family: "Lato", sans-serif;
    ${({ theme }) => theme.thone`font-size: 2em;`}
    ${({ theme }) => theme.phablet`
      font-size: 1.5em;
      line-height: 35px;
    `}
  }
  span {
    padding-right: 5px;
    padding-left: 5px;
    background: linear-gradient(
      transparent 10%,
      #495867 10%,
      #495867 100%,
      transparent 90%
    );
    color: #fff;
    font-weight: 300;
    text-align: justify;
  }
`;

export const CallToAction = styled.div`
  margin: 2em 0 4em 0;
  font-size: 15px;
  ${({ theme }) => theme.tablet`margin: 2em 0 4em 0;`}
  a {
    padding: 0 20px;
    height: 25px;
  }
  a::before,
  a::after {
    position: absolute;
    width: 25px;
    height: 1px;
    background: ${props => props.theme.text.dark};
    content: "";
    opacity: 0.2;
    transition: all 0.3s;
    pointer-events: none;
  }
  .dark-mode & a::before,
  .dark-mode & a::after {
    background: ${props => props.theme.text.light};
  }
  a:hover {
    opacity: 0.9;
  }
  a::before {
    top: 0;
    left: 0;
    transform: rotate(90deg);
    transform-origin: 0 0;
  }
  a::after {
    right: 0;
    bottom: 0;
    transform: rotate(90deg);
    transform-origin: 100% 0;
  }
  a:hover::before,
  a:hover::after,
  a:focus::before,
  a:focus::after {
    opacity: 1;
  }
  a:hover::before,
  a:focus::before {
    left: 50%;
    transform: rotate(0deg) translateX(-50%);
  }

  a:hover::after,
  a:focus::after {
    right: 50%;
    transform: rotate(0deg) translateX(50%);
  }
`;

export const Contact = styled.div`
  width: 20%;
  display: flex;
  justify-content: space-around;
  svg {
    height: 1.5em;
    width: 100%;
    fill: ${props => props.theme.text.dark};
  }
  .dark-mode & svg {
    fill: ${props => props.theme.text.light};
  }
  ${({ theme }) => theme.tablet`
    width: 40%;
    svg {
      height: 1.1em;
    }
  `}
`;
