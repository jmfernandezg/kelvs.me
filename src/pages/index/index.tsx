/* eslint-disable react/jsx-one-expression-per-line */
import Link from "next/link";
import Layout from "~/components/Layout";
import GmailSVG from "~/components/Icons/Gmail";
import GithubSVG from "~/components/Icons/Github";
import LinkedinSVG from "~/components/Icons/LinkedIn";
import { Container, HorizontalBar } from "~/components/PageStyles/common/style";
import {
  HeroWrapper,
  HeroOne,
  HeroTwo,
  CallToAction,
  Contact,
} from "~/components/PageStyles/index/style";

// https://www.justinewargnier.com/
// https://simpleicons.org/
// TODO: animate full stack dev text switch using react springs
export default function Home() {
  return (
    <Layout>
      <Container>
        <HeroWrapper>
          <HeroOne>
            <h1>
              Hello,&nbsp;
              <span>my name is Kelvin.</span>
            </h1>
            <HorizontalBar />
          </HeroOne>
          <HeroTwo>
            <h1>
              I&apos;m a <span>full&#8209;stack developer</span>
              &nbsp;and an open source enthusiast.
            </h1>
          </HeroTwo>
        </HeroWrapper>
        <CallToAction>
          <Link href="/about">
            <a>Learn more about me</a>
          </Link>
        </CallToAction>
        <Contact>
          <a href="https://github.com/kelvinabella">
            <GithubSVG />
          </a>
          <a href="https://www.linkedin.com/in/kelvinabella">
            <LinkedinSVG />
          </a>
          <a href="mailto:abellakelvin.ka@gmail.com">
            <GmailSVG />
          </a>
        </Contact>
      </Container>
    </Layout>
  );
}
