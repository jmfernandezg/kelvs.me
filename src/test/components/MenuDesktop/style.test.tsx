import React from "react";
import { render } from "~/test/utils/index";
import "jest-styled-components";

import {
  MenuDesktopContainer,
  LinkWrapper,
} from "~/components/MenuDesktop/style";

describe("MenuDesktop styles", () => {
  it("renders MenuDesktopContainer correctly", () => {
    const { container } = render(<MenuDesktopContainer />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders LinkWrapper correctly", () => {
    const { container } = render(<LinkWrapper isActive />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders LinkWrapper correctly", () => {
    const { container } = render(<LinkWrapper isActive={false} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
