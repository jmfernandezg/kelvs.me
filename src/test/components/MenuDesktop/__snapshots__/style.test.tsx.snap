// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuDesktop styles renders LinkWrapper correctly 1`] = `
.c0 a {
  text-align: center;
  margin: 0 2px;
  padding: 8px 32px;
  border-radius: 4px;
  background: rgba(0,122,255,0.1);
  color: #006be0;
}

.c0 a:hover {
  color: #007aff;
  background: rgba(0,122,255,0.1);
}

.dark-mode .c0 a {
  color: #fffffe;
  background: rgba(255,255,254,0.16);
}

.dark-mode .c0 a:hover {
  background: rgba(255,255,254,0.16);
}

<span
  class="c0"
/>
`;

exports[`MenuDesktop styles renders LinkWrapper correctly 2`] = `
.c0 a {
  text-align: center;
  margin: 0 2px;
  padding: 8px 32px;
  border-radius: 4px;
  background: none;
  color: #050505;
}

.c0 a:hover {
  color: #007aff;
  background: rgba(0,122,255,0.1);
}

.dark-mode .c0 a {
  color: #fffffe;
  background: none;
}

.dark-mode .c0 a:hover {
  background: rgba(255,255,254,0.16);
}

<span
  class="c0"
/>
`;

exports[`MenuDesktop styles renders MenuDesktopContainer correctly 1`] = `
.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

@media (max-width:768px) {
  .c0 {
    display: none;
  }
}

<div
  class="c0"
/>
`;
